### PR TITLE
Enable creating and editing projects based on ideas

### PR DESCRIPTION
**Pull request by daniel-patterson-ezcater**
## What did we change?

* Changed projects so they can be created from a dropdown of ideas
* Ensured projects can be edited
* Fixed some links to editing projects

## Why are we doing this?

We want to be able to create projects from ideas so we aren't duplicating info onto both models.

Here's a gif of this UI in action. Note: there isn't yet a `show` page for projects, so we don't yet have a place to show most of the information from the idea.

hackathon_projects

## Checklist
- [x] Automated Tests (check all that apply)
  - [ ] Unit
  - [x] Request/Integration
  - [ ] Feature/System
- [ ] Migrations Reviewed (Zero Downtime https://ezcater.atlassian.net/l/c/F1u4298i)

### CODE REVIEW

**Review by JessSumner — APPROVED** [review]

File: app/views/projects/_form.html.erb [hunk 1]
```diff
@@ -1,57 +1,27 @@
...
-        <%= f.text_field :name %>
+        <%= f.label "Which idea is project based on?" %><br />
+        <%= f.collection_select(
+            :idea_id,
+            available_ideas_for(@project),
+            :id,
+            :name,
+            { prompt: "Please select the idea this project will work on", include_blank: true },
+            { class: "full-width", required: true }
+        ) %>
```
Comment (tristandunn) [inline app/views/projects/_form.html.erb:11]: ❓ I wonder if this should be disabled once it's persisted?

File: spec/models/idea_spec.rb [hunk 2]
```diff
@@ -10,4 +10,13 @@
   it { is_expected.to validate_presence_of(:value_delivered) }
   it { is_expected.to validate_presence_of(:goal) }
   it { is_expected.to validate_presence_of(:hours_estimate) }
+
+  describe ".without_projects" do
+    it "returns just the ideas that do not have projects" do
+      FactoryBot.create_list(:project, 2).map(&:idea)
+      ideas_without_projects = FactoryBot.create_list(:idea, 2)
```
Comment (tristandunn) [inline spec/models/idea_spec.rb:17]: 🎱 Don't believe you need the `FactoryBot` portion since the syntax methods are included.

**Review by tristandunn — APPROVED** [review]
Couple of comments but looks good.

File: app/views/projects/_form.html.erb (hunk 1 above)
Comment (daniel-patterson-ezcater) [reply app/views/projects/_form.html.erb:11]: I was wondering that, too. I'll disable it for now.

File: spec/models/idea_spec.rb (hunk 2 above)
Comment (daniel-patterson-ezcater) [reply spec/models/idea_spec.rb:17]: Yep, I noticed that later and forgot to remove these. Will remove.

**Pull request merged**